griptape/engines/image_generation/prompt_image_generation_engine.py

```diff
@@ -0,0 +1,28 @@
+from attr import define, field, Factory
+from griptape.artifacts import ImageArtifact
+from griptape.drivers import (
+    Dalle2ImageGenerationDriver,
+    BaseImageGenerationDriver,
+)
+from griptape.engines.image_generation.base_image_generation_engine import (
+    BaseImageGenerationEngine,
+)
+from griptape.utils import J2
+
+
+@define
+class PromptImageGenerationEngine(BaseImageGenerationEngine):
```

Comment:
Should we make it a generic `ImageGenerationEngine` engine and add multiple methods for text-to-image and image-to-image modes? Or should we have different engines that encapsulate text-to-image (i.e., `PromptImageGenerationEngine`) and image-to-image?

Reply:
I've gone back and forth on it. This implementation represents the assumption that the interface and mechanics for text-to-image and image-to-image will be substantially different or that providers (and their drivers) might only provide a subset of that functionality. We could overcome each of those obstacles, just chose a path here.